nemo/collections/asr/models/ctc_bpe_models.py

```diff
@@ -48,24 +48,26 @@ def __init__(self, cfg: DictConfig, trainer=None):
         # Initialize a dummy vocabulary
         vocabulary = self.tokenizer.tokenizer.get_vocab()
 
-        # Set the new vocabulary
-        with open_dict(cfg):
-            # sidestepping the potential overlapping tokens issue in aggregate tokenizers
-            if self.tokenizer_type == "agg":
-                cfg.decoder.vocabulary = ListConfig(vocabulary)
-            else:
-                cfg.decoder.vocabulary = ListConfig(list(vocabulary.keys()))
-
-        # Override number of classes if placeholder provided
-        num_classes = cfg.decoder["num_classes"]
-
-        if num_classes < 1:
-            logging.info(
-                "\nReplacing placeholder number of classes ({}) with actual number of classes - {}".format(
-                    num_classes, len(vocabulary)
+        if "vocabulary" in cfg.decoder:
```

Comment:
This check is incorrect. For BPE models, vocabulary is set explicitly via setup_tokenizer(), so why is this check being done here? The default value of vocabulary is [] and is not even needed since it is injected from the models tokenizer.